library/x509_crt.c

```diff
@@ -1184,8 +1184,17 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
 
     while( ( entry = readdir( dir ) ) != NULL )
     {
-        mbedtls_snprintf( entry_name, sizeof entry_name, "%s/%s", path, entry->d_name );
-        //entry_name length is MBEDTLS_X509_MAX_FILE_PATH_LEN  and it includes both path and filename
+        if( strlen(path) + strlen(entry->d_name) + 1 < MBEDTLS_X509_MAX_FILE_PATH_LEN )
+        {
+            mbedtls_snprintf( entry_name, sizeof entry_name, "%s/%s", path, entry->d_name );         
```

Comment:
Please remove the trailing whitespace.

As it happens we'll soon have a script that rejects all trailing whitespace.